client.go

```diff
@@ -10,58 +10,97 @@ import (
 	"github.com/gopcua/opcua/uasc"
 )
 
+var DefaultSessionConfig = uasc.NewClientSessionConfig(
+	[]string{"en-US"},
+	&ua.AnonymousIdentityToken{PolicyID: "open62541-anonymous-policy"},
+)
+
 // Client is a high-level client for an OPC/UA server.
 // It establishes a secure channel and a session.
 type Client struct {
-	Addr string
+	// Endpoint is the endpoint URL the client connects to.
+	Endpoint string
```

Comment:
This is definitely more clear, however, my changes will probably still require to pass in the actual endpoint description or some variant of it and 'Endpoint' is a good name for that field, too.  There is a lot of information within the endpoint description that helps the stack do its job (security policy, certificate info, authorization tokens, etc)

Reply:
Shouldn't these be part of the `Config` object as they are right now? Otherwise, you'd use `EndpointDescription` as a convenient container but never pass this directly to some other function. Gut-feel from my side is not to use the protocol objects for configuration unless they are directly required for the configuration.